src/pages/HomePage.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { W3mButton } from '@web3modal/wagmi-react-native';
+import { Alert, Button, StyleSheet, Text, View } from 'react-native';
+import {
+  useAccount,
+  useBalance,
+  useBlockNumber,
+  useFeeData,
+  useNetwork,
+  usePrepareSendTransaction,
+  useSendTransaction,
+  useSignMessage,
+  useWaitForTransaction,
+} from 'wagmi';
+import { parseEther } from 'viem';
+import { polygonMumbai } from 'wagmi/chains';
+
+export default function HomePage() {
+  const { chain } = useNetwork();
+  const { data: blockNumber } = useBlockNumber();
+  const { data: feeData } = useFeeData({
+    chainId: chain?.id,
+    formatUnits: 'ether',
+  });
+  const { address, isConnected } = useAccount();
+  const { data: balance } = useBalance({ address });
+  const { data: signedHash, signMessage } = useSignMessage({
+    message: 'Sign this message to prove you are the owner of this wallet',
+  });
+
+  // Send test MATIC on Polygon Mumbai testnet
```

Comment:
Would it make sense to extract the transaction part to separate component. THis page is getting quite busy with different concerns: chain data, action data, signing message, send tx. It would be good to split it into more managable parts

Reply:
yeah, good idea

Reply:
Done! Created a new component `WalletActions` that holds signing, sending tx, and watching for tx success